### ISSUE TITLE
web pages are loading very slow?

### ISSUE THREAD

**Issue by hspark15429** [DanielnetoDotCom/YouPHPTube#292]
Not sure why but every page is loading super slow. Uploading videos seems fast though. I'm using AWS ec2 medium server.

**Comment by DanielnetoDotCom** [comment]
can you share a link to us?

**Comment by SeanMollet** [comment]
I've tracked this one down. In navbar.php on line 11, it uses file_get_contents to retrieve a php file from itself. That requires two things, allow_url_fopen must be On, and the server must be able to connect to itself by domain. I run my webserver behind a NAT and an https transparent proxy. So, it can't usually connect to itself.
I fixed it the quick and dirty way by adding the domain name to /etc/hosts pointing to the internal IP of the webserver.
I believe Amazon does the same thing regarding natting of the webserver.

Here's a link to a possible alternative way to handle this that wouldn't require self connection:
https://stackoverflow.com/questions/5874466/assign-output-of-execution-of-php-script-to-a-variable

It's abusing the output buffer to execute a sub-script and capture the contents.
